blueocean-i18n/src/main/java/io/jenkins/blueocean/i18n/BlueI18n.java

```diff
@@ -0,0 +1,394 @@
+        }
+
+        return null;
+    }
+
+    @CheckForNull
+    static BundleParams getBundleParameters(String restOfPath) {
+        if (restOfPath == null || restOfPath.length() == 0) {
+            return null;
+        }
+
+        String[] pathTokens = restOfPath.split("/");
+        List<String> bundleParameters = new ArrayList<>();
+
+        for (String pathToken : pathTokens) {
+            if (pathToken.length() > 0) {
+                bundleParameters.add(pathToken);
+            }
+        }
+
+        if (bundleParameters.size() != 3 && bundleParameters.size() != 4) {
+            return null;
+        }
+
+        BundleParams bundleParams = new BundleParams(
+            bundleParameters.get(0),
+            bundleParameters.get(1),
+            bundleParameters.get(2)
+        );
+        if (bundleParameters.size() == 4) {
+            // https://www.w3.org/International/questions/qa-lang-priorities
+            // in case we have regions/countries in the language query parameter
+            String locale = bundleParameters.get(3);
+            String[] localeTokens = locale.split("-|_");
+            bundleParams.language = localeTokens[0];
+            if (localeTokens.length > 1) {
+                bundleParams.country = localeTokens[1];
+                if (localeTokens.length > 2) {
+                    bundleParams.variant = localeTokens[2];
+                }
+            }
+        }
+
+        return bundleParams;
+    }
+
+    @CheckForNull
+    static PluginWrapper getPlugin(String pluginName) {
+        Jenkins jenkins = Jenkins.getInstance();
+        return jenkins.getPluginManager().getPlugin(pluginName);
+    }
+
+    static class BundleParams {
+        final String pluginName;
+        final String pluginVersion;
+        final String bundleName;
+        String language;
+        String country;
+        String variant;
+        private PluginWrapper plugin;
+
+        BundleParams(String pluginName, String pluginVersion, String bundleName) {
+            this.pluginName = pluginName;
+            this.pluginVersion = pluginVersion;
+            this.bundleName = bundleName;
+        }
+
+        @CheckForNull
+        Locale getLocale() {
+            if (language != null && country != null && variant != null) {
+                return new Locale(language, country, variant);
+            } else if (language != null && country != null) {
+                return new Locale(language, country);
+            } else if (language != null) {
+                return new Locale(language);
+            } else {
+                return null;
+            }
+        }
+
+        // We declare the plugin to be a release version iff
+        // the version string contains only numerics.
+        private static Pattern RELEASE_VERSION_PATTERN = Pattern.compile("[\\d/.]{3,}");
+        boolean isReleaseVersion(String version) {
+            return RELEASE_VERSION_PATTERN.matcher(version).matches();
```

Comment:
Yeah, but it wouldn't work for version strings like `1.0.0-b13-SNAPSHOT (private-fd50c615-tfennelly)`